src/Microsoft.Health.Dicom.Functions/DataCleanup/DataCleanupDurableFunction.Activity.cs

```diff
@@ -0,0 +1,115 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Dicom.Core.Features.Model;
+using Microsoft.Health.Dicom.Core.Models;
+using Microsoft.Health.Dicom.Functions.DataCleanup.Models;
+
+namespace Microsoft.Health.Dicom.Functions.DataCleanup;
+
+public partial class DataCleanupDurableFunction
+{
+    ///<summary>
+    /// Asynchronously retrieves the next set of instance batches based on the configured options.
+    /// </summary>
+    /// <param name="arguments">The options for configuring the batches.</param>
+    /// <param name="logger">A diagnostic logger.</param>
+    /// <returns>
+    /// A task representing the asynchronous get operation. The value of its <see cref="Task{TResult}.Result"/>
+    /// property contains a list of batches as defined by their smallest and largest watermark.
+    /// </returns>
+    /// <exception cref="ArgumentNullException">
+    /// <paramref name="arguments"/> or <paramref name="logger"/> is <see langword="null"/>.
+    /// </exception>
+    [FunctionName(nameof(GetInstanceBatchesByTimeStampAsync))]
+    public Task<IReadOnlyList<WatermarkRange>> GetInstanceBatchesByTimeStampAsync(
```

Comment:
Did you say offline you were going to make this sort of thing more generic? If so, should this activity live outside this particular class?

Reply:
I am thinking of fixing the customer and refactoring later when I get time. What do you think?

Reply:
Yeah, I think we can make it generic afterwards